File: MoSmartPark/UI/mosmartpark_desktop/lib/model/user_preferences.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_preferences.g.dart';

@JsonSerializable()
class UserPreferences {
  final int id;
  final int userId;
  final String themeMode;
  final int? defaultParkingZoneId;
  final String? defaultParkingZoneName;
  final bool notifyReviews;
  final bool notifyReservations;
  final bool notifyCars;
  final DateTime? updatedAt;

  UserPreferences({
    this.id = 0,
    this.userId = 0,
    this.themeMode = 'system',
    this.defaultParkingZoneId,
    this.defaultParkingZoneName,
    this.notifyReviews = true,
    this.notifyReservations = true,
    this.notifyCars = true,
    this.updatedAt,
  });

  factory UserPreferences.fromJson(Map<String, dynamic> json) =>
      _$UserPreferencesFromJson(json);
  Map<String, dynamic> toJson() => _$UserPreferencesToJson(this);
}

File: MoSmartPark/MoSmartPark.Services/Database/UserPreferences.cs
using System.ComponentModel.DataAnnotations;

namespace MoSmartPark.Services.Database
{
    public class UserPreferences
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        [MaxLength(20)]
        public string ThemeMode { get; set; } = "system";

        public int? DefaultParkingZoneId { get; set; }

        public bool NotifyReviews { get; set; } = true;

        public bool NotifyReservations { get; set; } = true;

        public bool NotifyCars { get; set; } = true;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public User User { get; set; } = null!;
        public ParkingZone? DefaultParkingZone { get; set; }
    }
}
